packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js

```diff
@@ -184,6 +205,32 @@ class TaintTrackingPlugin extends SourceIastPlugin {
     this.taintHeaders(req.headers, iastContext)
     this.taintUrl(req, iastContext)
   }
+
+  _taintDatabaseResult (result, dbOrigin, iastContext = getIastContext(storage.getStore()), name) {
+    if (!iastContext) return result
+
+    if (this._rowsToTaint === 0) return
+
+    if (Array.isArray(result)) {
+      for (let i = 0; i < result.length && i < this._rowsToTaint; i++) {
+        const nextName = name ? `${name}.${i}` : '' + i
```

Comment:
```suggestion
  _taintDatabaseResult (result, dbOrigin, iastContext = getIastContext(storage.getStore()), name = '') {
    if (!iastContext) return result

    if (this._rowsToTaint === 0) return

    if (Array.isArray(result)) {
      for (let i = 0; i < result.length && i < this._rowsToTaint; i++) {
        const nextName = `${name}.${i}`
```

Reply:
I don't like the change, and with this change, if name is empty string, the nextName will be `.1` in an array, and not just `1`. 
You are adding an extra `.` here. And it would be weird, if I apply same change in line 224
```javascript
for (const key in result) {
  const nextName = name ? `${name}.${key}` : key
  result[key] = this._taintDatabaseResult(result[key], dbOrigin, iastContext, nextName)
}
```

Reply:
ah true.
another option could be `name = 'result'`